examples/check.rs

```diff
@@ -68,6 +71,82 @@ fn run(args: &Args) -> CargoResult<()> {
         manifests.collect()
     };
 
+    let lockfile_path = args
+        .lockfile_path
+        .clone()
+        .unwrap_or(current_dir()?.join("Cargo.lock"));
+
+    let lockfile = {
+        if lockfile_path.is_file() {
+            let mut cmd = Command::new("cargo");
+            cmd.args(["run", "plumbing", "read-lockfile"])
+                .arg("--lockfile-path")
+                .arg(&lockfile_path)
+                .stdout(Stdio::piped());
+
+            let mut child = cmd.spawn().expect("failed to run read-manifest");
+            let mut stdout = child.stdout.take().expect("failed to get stdout");
+
+            let mut buffer = Vec::new();
+            stdout
+                .read_to_end(&mut buffer)
+                .expect("failed to read stdout");
+
+            child.wait().expect("failed to wait for read-manifest");
```

Comment:
Why do we read like this rather than calling `.output()`?  Its not like we're reading incrementally since we `wait`